Mage/src/main/java/mage/cards/decks/importer/MtgaImporter.java

```diff
@@ -22,38 +22,52 @@ public class MtgaImporter extends PlainTextDeckImporter {
             "(\\p{Digit}+)" +
                     "\\p{javaWhitespace}+" +
                     "(" + CARD_NAME_PATTERN.pattern() + ")" +
-                    "\\p{javaWhitespace}+" +
-                    "\\((\\p{Alnum}+)\\)" +
-                    "\\p{javaWhitespace}+" +
-                    "(\\p{Digit}+)");
+                    "(?:\\p{javaWhitespace}+\\()?" +
+                    "(\\p{Alnum}+)?" +
+                    "(?:\\)\\p{javaWhitespace}+)?" +
+                    "(\\p{Graph}+)?");
 
     private final CardLookup lookup = getCardLookup();
     private boolean sideboard = false;
 
     @Override
     protected void readLine(String line, DeckCardLists deckList, FixedInfo fixedInfo) {
+        
+        line = line.trim();
+        
+        if (line.equals("Deck")) {
+            return;
+        }
 
-        if (line.trim().equals("")) {
+        if (line.equals("Sideboard") || line.equals("")) {
             sideboard = true;
             return;
         }
 
         Matcher m = MTGA_PATTERN.matcher(CardNameUtil.normalizeCardName(line));
-        if (m.matches()) {
-            int count = Integer.parseInt(m.group(1));
-            String name = m.group(2);
+        
+        if (!m.matches()) {
+            sbMessage.append("Error reading '").append(line).append("'\n");
+            return;
+        }
+        
+        Optional<CardInfo> found;
+        int count = Integer.parseInt(m.group(1));
+        String name = m.group(2);    
+        if (m.group(3) != null && m.group(4) != null) {
```

Comment:
I know you didn't create it, but could you put a more descriptive variable name than `m`

Reply:
Changed "m" to "pattern"